test/finite_differencing.jl

```diff
@@ -1,58 +1,78 @@
+import Nabla: ∇, compute_Dv, approximate_Dv
 @testset "finite_differencing" begin
+
     let
         # Define a dummy test function.
         foo(x::∇Real) = 5x
-        foo(x::Vector{<:∇Real}) = 10x
-        foo(x::Matrix{<:∇Real}) = 15x
+        foo(x::∇Real, y::∇Real) = 5x + 6y
+        foo(x::Vector{<:∇Real}, y::Vector{<:∇Real}) = 10x + 11y
+        foo(x::Matrix{<:∇Real}, y::Matrix{<:∇Real}) = 15x + 16y
 
         # Create sensitivity intercepts.
         @explicit_intercepts foo Tuple{∇Real}
-        @explicit_intercepts foo Tuple{Vector{<:∇Real}}
-        @explicit_intercepts foo Tuple{Matrix{<:∇Real}}
+        @explicit_intercepts foo Tuple{∇Real, ∇Real}
+        @explicit_intercepts foo Tuple{Vector{<:∇Real}, Vector{<:∇Real}}
+        @explicit_intercepts foo Tuple{Matrix{<:∇Real}, Matrix{<:∇Real}}
 
         # Define sensitivity implementations.
-        Nabla.∇(::typeof(foo), ::Type{Arg{1}}, p, y, ȳ, x::∇Real) = 5ȳ
-        Nabla.∇(::typeof(foo), ::Type{Arg{1}}, p, y, ȳ, x::Vector{<:∇Real}) = 10ȳ
-        Nabla.∇(::typeof(foo), ::Type{Arg{1}}, p, y, ȳ, x::Matrix{<:∇Real}) = 15ȳ
+        const _Vec = Vector{<:∇Real}
+        const _Mat = Matrix{<:∇Real}
+        Nabla.∇(::typeof(foo), ::Type{Arg{1}}, p, z, z̄, x::∇Real) = 5z̄
```

Comment:
In principle I completely agree. I was doing this, but I got a weird method error that disappeared if I just qualified it directly, because for some reason it creates a new `∇` function. Furthermore, if I tried to move the `import` inside the `testset` or `let` blocks I get a weird `syntax: malformed "import" statement` error.